test/distributed/org/apache/cassandra/distributed/impl/AbstractCluster.java

```diff
@@ -332,29 +333,12 @@ private IInvokableInstance newInstance()
             ++generation;
             IClassTransformer transformer = classTransformer == null ? null : classTransformer.initialise();
             ClassLoader classLoader = new InstanceClassLoader(generation, config.num(), version.classpath, sharedClassLoader, sharedClassPredicate, transformer);
-            Consumer<Throwable> stabilityInspector;
-            {
-                try
-                {
-                    Class<?> owner = classLoader.loadClass(JVMStabilityInspector.class.getName());
-                    Method method = owner.getMethod("inspectThrowable", Throwable.class);
```

Comment:
`uncaughtException` was needed to have correct logging, using this method hid logs

Reply:
could you elaborate?

Reply:
the main reason is that this logic is trying to pull `JVMStabilityInspector` into the JVM-Dtest api which makes things harder for upgrade testing... now any change to that class must be coordinated cross every branch, and you can no longer test *released* versions... 

I don't see a clear reason to bring this internal API into the public api when we already have `instance.uncaughtException` that solves this issue for us.

> could you elaborate?

`uncaughtException` comes with extra logging so when you look at the logs you see these happening, when you call `inspectThrowable` those logs don't happen which makes debugging a bit harder.  I know that an error happened, but "where" did it happen?  🤷

Reply:
So why not add just that extra logging in the exception handler we register? This would also be clearer. My problem with the current and prior approach is that we're invoking this all on _another thread_ via a weird self-referential loop. Passing the Cluster through to each instance to only lookup the instance itself again is confusing to me, and unnecessary. 

We can if you like define a static method for this that we mark as requiring stability for API purposes, or if you really want we can jump through some extra hoops to invoke the IInstance methods directly without the indirection via Cluster and additional threads.